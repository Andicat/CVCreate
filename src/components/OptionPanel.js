import React from 'react';
import PropTypes from 'prop-types';
import Option from './Option';
import {CV_ID} from './utils';

import {connect} from 'react-redux';
import {cvStyle_update,
    cvBlock_sendBack,
    cvBlock_copy,
    cvBlock_setSize,
    cvBlocks_align,
    cvBlocks_alignSize,
    cvBlocks_distribute,
    cvBlocks_group,
    cvBlock_ungroup, 
    cvBlock_lock,
    cvBlock_setLink,
    templates_add } from '../redux/cvDataAC';
import {getAutoSize} from './utils';

class OptionPanel extends React.PureComponent {

    static propTypes = {
        styleToEdit: PropTypes.object,
        activeElementId: PropTypes.string,
        activeBlockOptions: PropTypes.object,
        activeBlocksId: PropTypes.array,    
        activeBlockDOM: PropTypes.object, 
        stylePage: PropTypes.object,   
    };

    BLOCK_ACTION = [
        'lock',
        'autosize',
        'back',
        'copy',
        'link',
        'save',
    ];

    BLOCKS_ACTION = [
        'align_top',
        'align_bottom',
        'align_left',
        'align_right',
        'align_vertical',
        'align_horisontal',
        'distribute_vertical',
        'distribute_horisontal',
        'align_width',
        'align_height',
        'group',
    ];

    setStyle = (blockId,optionName,value) => {
        this.props.dispatch(cvStyle_update(blockId,optionName,value));
    }

    setAction = (blockId,optionName,value) => {
        switch (optionName) {
            case 'align_top': {
                this.props.dispatch(cvBlocks_align('top'));
                break;
            }
            case 'align_bottom': {
                this.props.dispatch(cvBlocks_align('bottom'));
                break;
            }
            case 'align_left': {
                this.props.dispatch(cvBlocks_align('left'));
                break;
            }
            case 'align_right': {
                this.props.dispatch(cvBlocks_align('right'));
                break;
            }
            case 'align_vertical': {
                this.props.dispatch(cvBlocks_align('vertical'));
                break;
            }
            case 'align_horisontal': {
                this.props.dispatch(cvBlocks_align('horisontal'));
                break;
            }
            case 'distribute_vertical': {
                this.props.dispatch(cvBlocks_distribute('vertical'));
                break;
            }
            case 'distribute_horisontal': {
                this.props.dispatch(cvBlocks_distribute('horisontal'));
                break;
            }
            case 'align_width': {
                this.props.dispatch(cvBlocks_alignSize('width'));
                break;
            }
            case 'align_height': {
                this.props.dispatch(cvBlocks_alignSize('height'));
                 break;
            }
            case 'group': {
                this.props.dispatch(cvBlocks_group());
                this.setState({tooltip:null});
                break;
            }
            case 'lock': {
                this.props.dispatch(cvBlock_lock(blockId));
                break;
            }
            case 'autosize': {
                if (this.props.activeBlockDOM) {
                    let sizesAuto = getAutoSize(this.props.activeBlockDOM);
                    this.props.dispatch(cvBlock_setSize(blockId,sizesAuto.height,sizesAuto.width));            
                }
                break;
            }
            case 'back': {
                this.props.dispatch(cvBlock_sendBack(blockId));
                this.setState({tooltip:null});
                break;
            }
            case 'copy': {
                this.props.dispatch(cvBlock_copy(blockId));
                this.setState({tooltip:null});
                break;
            }
            case 'ungroup': {
                this.props.dispatch(cvBlock_ungroup(blockId));
                this.setState({tooltip:null});
                break;
            }
            case 'link': {
                this.props.dispatch(cvBlock_setLink(blockId,value));
                break;
            }
            case 'save': {
                this.props.dispatch(templates_add(blockId));
                break;
            }
            default:
                //console.log('action', optionName, value);
                return;
        }
    }

    render () {
        //debugger
        //console.log('render option panel', this.props);
        let codeElementOptions = null;
        let codeBlockOptions = null;
        let codeBlocksOptions = null;
        let codePageOptions = null;
        let blockAction = [...this.BLOCK_ACTION];
        
        if (this.props.activeBlocksId.length > 1) { //few active blocks
            codeBlocksOptions = this.BLOCKS_ACTION.map( (a,i) => (
                <Option key={i} optionName={a} cbOnChange={this.setAction}/>));
        } else if (this.props.activeBlockOptions) { //one active block
            if (this.props.activeBlockOptions.group) {
                blockAction.push('ungroup');
            } else {
               // blockAction.push('autosize');
            }
            codeBlocksOptions = blockAction.map( (a,i) => {
                let value = null;
                if (a==='lock') {
                    value = !!this.props.activeBlockOptions.lock;
                } else if (a==='link') {
                    value = this.props.activeBlockOptions.link;
                }
                return <Option key={'block-' + i} optionName={a} optionValue={value} blockId={this.props.activeBlockOptions.id} cbOnChange={this.setAction}/>;
            });
            if (this.props.activeElementId) { //active element
                
                let styles = Object.keys(this.props.styleToEdit).sort();
                codeElementOptions = styles.map( (s,i) => (
                    <Option key={'option-' + i} optionName={s} optionValue={this.props.styleToEdit[s]} blockId={this.props.activeBlockOptions.id} cbOnChange={this.setStyle}/>));
            };
        } else if (this.props.stylePage) { //non active block, but active page
            let styles = Object.keys(this.props.stylePage).sort();
            codePageOptions = styles.map( (s,i) => (
                <Option key={i} optionName={s} optionValue={this.props.stylePage[s]} blockId={CV_ID} cbOnChange={this.setStyle}/>));
        }

        return (
            <div className='option-panel'>
                {((codePageOptions && !codeBlocksOptions) || codeElementOptions) && 
                    <div className='option-panel__group option-panel__group--style'>
                        {codePageOptions}
                        {codeElementOptions}
                    </div>
                }
                {codeBlocksOptions && 
                    <div className='option-panel__group option-panel__group--blocks'>
                        {codeBlocksOptions}
                    </div>
                }
                {codeBlockOptions && 
                    <div className='option-panel__group option-panel__group--block'>
                        {codeBlockOptions}
                    </div>
                }
            </div>
        );
    }
}

const mapStateToProps = function (state) {
    return {
        stylePage: state.cvData.stylePage,
        styleToEdit: state.cvData.styleToEdit,
        activeElementId: state.cvData.activeElementId,
        activeBlocksId: state.cvData.activeBlocksId,
        activeBlockDOM: state.cvData.activeBlockDOM,
    };
};

export default connect(mapStateToProps)(OptionPanel);
