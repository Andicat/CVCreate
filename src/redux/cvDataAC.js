const CV_BLOCK_ADD = 'CV_BLOCK_ADD';
const CV_BLOCK_DELETE = 'CV_BLOCK_DELETE';
const CV_BLOCK_MOVE = 'CV_BLOCK_MOVE';
const CV_BLOCK_RESIZE = 'CV_BLOCK_RESIZE';
const CV_BLOCK_ACTIVATE = 'CV_BLOCK_ACTIVATE';
const CV_BLOCK_ACTIVATE_MULTI = 'CV_BLOCK_ACTIVATE_MULTI';
const CV_BLOCK_COPY = 'CV_BLOCK_COPY';
const CV_BLOCK_SET_SIZE = 'CV_BLOCK_SET_SIZE';
const CV_ELEMENT_ACTIVATE = 'CV_ELEMENT_ACTIVATE';
const CV_STYLE_UPDATE = 'CV_STYLE_UPDATE';
const CV_TEXT_UPDATE = 'CV_TEXT_UPDATE';
const CV_BLOCK_SEND_BACK = 'CV_BLOCK_SEND_BACK';
const CV_BLOCKS_ALIGN_TOP = 'CV_BLOCKS_ALIGN_TOP';
const CV_BLOCKS_ALIGN_BOTTOM = 'CV_BLOCKS_ALIGN_BOTTOM';
const CV_BLOCKS_ALIGN_LEFT = 'CV_BLOCKS_ALIGN_LEFT';
const CV_BLOCKS_ALIGN_RIGHT = 'CV_BLOCKS_ALIGN_RIGHT';
const CV_BLOCKS_ALIGN_VERTICAL = 'CV_BLOCKS_ALIGN_VERTICAL';
const CV_BLOCKS_ALIGN_HORISONTAL = 'CV_BLOCKS_ALIGN_HORISONTAL';
const CV_BLOCKS_DISTRIBUTE_VERTICAL = 'CV_BLOCKS_DISTRIBUTE_VERTICAL';
const CV_BLOCKS_DISTRIBUTE_HORISONTAL = 'CV_BLOCKS_DISTRIBUTE_HORISONTAL';
const CV_BLOCKS_ALIGN_WIDTH = 'CV_BLOCKS_ALIGN_WIDTH';
const CV_BLOCKS_ALIGN_HEIGHT = 'CV_BLOCKS_ALIGN_HEIGHT';
const CV_BLOCKS_GROUP = 'CV_BLOCKS_GROUP';
const CV_BLOCK_UNGROUP = 'CV_BLOCK_UNGROUP';
const CV_BLOCK_LOCK = 'CV_BLOCK_LOCK';
const CV_LOAD = 'CV_LOAD';
const CV_INIT = 'CV_INIT';
const CV_SET_USER = 'CV_SET_USER';
const CV_SET_LINK = 'CV_SET_LINK';
const CV_BLOCK_LINK = 'CV_BLOCK_LINK';
const TEMPLATE_ADD = 'TEMPLATE_ADD';

const cvBlock_add = function(block) {
    return {
        type: CV_BLOCK_ADD,
        block:block,
    };
}

const cvBlock_delete = function(blockId) {
    return {
        type: CV_BLOCK_DELETE,
        blockId:blockId,
    };
}

const cvBlock_move = function(blockId,shiftTop,shiftLeft) {
    return {
        type: CV_BLOCK_MOVE,
        blockId:blockId,
        shiftTop: shiftTop,
        shiftLeft: shiftLeft,
    };
}

const cvBlock_resize = function(blockId,shiftHeight,shiftWidth) {
    return {
        type: CV_BLOCK_RESIZE,
        blockId:blockId,
        shiftHeight: shiftHeight,
        shiftWidth: shiftWidth,
    };
}

const cvBlock_activate = function(blockId,target) {
    return {
        type: CV_BLOCK_ACTIVATE,
        blockId:blockId,
        target:target,
    };
}

const cvElement_activate = function(style,elementId) {
    return {
        type: CV_ELEMENT_ACTIVATE,
        elementId:elementId,
        style:style,
    };
}

const cvStyle_update = function(blockId,styleName,styleValue) {
    return {
        type: CV_STYLE_UPDATE,
        blockId: blockId,
        styleName: styleName,
        styleValue: styleValue,
    };
}

const cvElement_textUpdate = function(blockId,textValue) {
    return {
        type: CV_TEXT_UPDATE,
        blockId: blockId,
        textValue: textValue,
    };
}

const cvBlock_sendBack = function(blockId) {
    return {
        type: CV_BLOCK_SEND_BACK,
        blockId: blockId,
    };
}

const cvBlock_copy = function(blockId) {
    return {
        type: CV_BLOCK_COPY,
        blockId: blockId,
    };
}

const cvBlock_setSize = function(blockId,height,width) {
    return {
        type: CV_BLOCK_SET_SIZE,
        blockId: blockId,
        width:width,
        height:height,
    };
}

const cvBlock_activateMulti = function(blockId) {
    return {
        type: CV_BLOCK_ACTIVATE_MULTI,
        blockId: blockId,
    };
}

const cvBlocks_align = function(mode) {
    switch (mode) {
        case 'top':
            return { type: CV_BLOCKS_ALIGN_TOP };
        case 'bottom':
            return { type: CV_BLOCKS_ALIGN_BOTTOM };
        case 'left':
            return { type: CV_BLOCKS_ALIGN_LEFT };
        case 'right':
            return { type: CV_BLOCKS_ALIGN_RIGHT };
        case 'vertical':
            return { type: CV_BLOCKS_ALIGN_VERTICAL };
        case 'horisontal':
            return { type: CV_BLOCKS_ALIGN_HORISONTAL };
        default:
    }
}

const cvBlocks_distribute = function(mode) {
    switch (mode) {
        case 'vertical':
            return { type: CV_BLOCKS_DISTRIBUTE_VERTICAL };
        case 'horisontal':
            return { type: CV_BLOCKS_DISTRIBUTE_HORISONTAL };
        default:
    }
}

const cvBlocks_alignSize = function(mode) {
    switch (mode) {
        case 'width':
            return { type: CV_BLOCKS_ALIGN_WIDTH };
        case 'height':
            return { type: CV_BLOCKS_ALIGN_HEIGHT };
        default:
    }
}

const cvBlocks_group = function() {
    return { type: CV_BLOCKS_GROUP };
}

const cvBlock_ungroup = function(blockId) {
    return { 
        type: CV_BLOCK_UNGROUP,
        blockId: blockId,
    };
}

const cvBlock_lock = function(blockId) {
    return { 
        type: CV_BLOCK_LOCK,
        blockId: blockId,
    };
}

const cv_load = function(blocks,style) {
    return { 
        type: CV_LOAD,
        blocks:blocks,
        style:style,
    };
}

const cv_init = function(blocks,style,user,link,templatesData,templatesUser) {
    return { 
        type: CV_INIT,
        blocks:blocks,
        style:style,
        user:user,
        link:link,
        templatesData:templatesData,
        templatesUser:templatesUser,
    };
}

const cv_setUser = function(user) {
    return { 
        type: CV_SET_USER,
        user:user,
    };
}

const cv_setLink = function(link) {
    return { 
        type: CV_SET_LINK,
        link:link,
    };
}

const cvBlock_setLink = function(blockId,linkValue) {
    return { 
        type: CV_BLOCK_LINK,
        blockId:blockId,
        linkValue:linkValue,
    };
}

const templates_add = function(blockId) {
    return {
        type: TEMPLATE_ADD,
        blockId: blockId,
    };
}

export {
    cvBlock_add, CV_BLOCK_ADD,
    cvBlock_delete, CV_BLOCK_DELETE,
    cvBlock_move, CV_BLOCK_MOVE,
    cvBlock_resize, CV_BLOCK_RESIZE,
    cvBlock_activate, CV_BLOCK_ACTIVATE,
    cvBlock_activateMulti, CV_BLOCK_ACTIVATE_MULTI,
    cvBlock_sendBack, CV_BLOCK_SEND_BACK,
    cvBlock_copy, CV_BLOCK_COPY,
    cvBlock_setSize, CV_BLOCK_SET_SIZE,
    cvBlock_lock, CV_BLOCK_LOCK,
    cvBlock_setLink, CV_BLOCK_LINK,
    cvBlocks_align, CV_BLOCKS_ALIGN_TOP, CV_BLOCKS_ALIGN_BOTTOM, CV_BLOCKS_ALIGN_LEFT, CV_BLOCKS_ALIGN_RIGHT, CV_BLOCKS_ALIGN_VERTICAL, CV_BLOCKS_ALIGN_HORISONTAL,
    cvBlocks_distribute, CV_BLOCKS_DISTRIBUTE_VERTICAL, CV_BLOCKS_DISTRIBUTE_HORISONTAL,
    cvBlocks_alignSize, CV_BLOCKS_ALIGN_WIDTH, CV_BLOCKS_ALIGN_HEIGHT,
    cvBlocks_group, CV_BLOCKS_GROUP, 
    cvBlock_ungroup, CV_BLOCK_UNGROUP,
    cvElement_activate, CV_ELEMENT_ACTIVATE,
    cvStyle_update, CV_STYLE_UPDATE,
    cvElement_textUpdate, CV_TEXT_UPDATE,
    cv_load, CV_LOAD,
    cv_init, CV_INIT, 
    cv_setUser, CV_SET_USER,
    cv_setLink, CV_SET_LINK,
    templates_add, TEMPLATE_ADD
}
