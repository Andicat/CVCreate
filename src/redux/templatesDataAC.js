const TEMPLATE_LOAD = 'TEMPLATE_LOAD';

const templates_load = function(data) {
    return {
        type: TEMPLATE_LOAD,
        data: data,
    };
}

export { templates_load, TEMPLATE_LOAD }
