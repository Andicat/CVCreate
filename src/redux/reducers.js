import { combineReducers } from 'redux';

import cvDataReducer from "./cvDataReducer";

let combinedReducer=combineReducers({
    cvData: cvDataReducer,
});

export default combinedReducer;
