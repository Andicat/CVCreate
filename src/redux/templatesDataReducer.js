import { TEMPLATE_LOAD } from './templatesDataAC';

const initState = {
    templatesArr: [],
    imageUrl: null,
}

function templatesDataReducer(state = initState, action) {

    switch (action.type) {
        //load
        case TEMPLATE_LOAD: {
            let newState = {templatesArr:action.data.templates, imageUrl:action.data.image};
            return newState;
        }
        default:
            return state;
    }
}

export default templatesDataReducer;
