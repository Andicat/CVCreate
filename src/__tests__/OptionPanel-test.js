

import {renderComponentWithStore, storeWithData} from './utils-test';
import {cvStyle_update,cvBlock_lock} from '../redux/cvDataAC';

import OptionPanel from '../components/OptionPanel';

let component;

let props = {activeBlockOptions: {id: 4, lock: false, link: '', group: false,}};

describe('OptionPanel render&actions', () => {

    it('Изменение цвета текста', () => {
        storeWithData.clearActions();
        component = renderComponentWithStore(OptionPanel,storeWithData,props);
        expect(component.toJSON()).toMatchSnapshot();
        let optionColor = component.root.find( el => el.props.className==='option__color');
        const evt = {'target': {'value': '#666'}};
        optionColor.props.onChange(evt);
        const expectedActions = [
            cvStyle_update(props.activeBlockOptions.id,'s01_color','#666'),
        ];
        expect(storeWithData.getActions()).toEqual(expectedActions);
    });

    it('Блокирование расположения блока', () => {
        storeWithData.clearActions();
        component = renderComponentWithStore(OptionPanel,storeWithData,props);
        expect(component.toJSON()).toMatchSnapshot();
        let optionLock = component.root.find( el => el.props.className==='option__checkbox option__checkbox--lock');
        const evt = {'target': {'checked': true}};
        optionLock.props.onChange(evt);
        const expectedActions = [
            cvBlock_lock(props.activeBlockOptions.id),
        ];
        expect(storeWithData.getActions()).toEqual(expectedActions);

    });
});
