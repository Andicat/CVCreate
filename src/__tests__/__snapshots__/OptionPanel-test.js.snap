// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionPanel render&actions Блокирование расположения блока 1`] = `
<div
  className="option-panel"
>
  <div
    className="option-panel__group option-panel__group--style"
  >
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <select
        className="option__select"
        data-tooltip={true}
        onChange={[Function]}
        value="Roboto"
      >
        <option
          style={
            Object {
              "fontFamily": "PTSans",
            }
          }
        >
          PTSans
        </option>
        <option
          style={
            Object {
              "fontFamily": "Roboto",
            }
          }
        >
          Roboto
        </option>
        <option
          style={
            Object {
              "fontFamily": "Helvetica",
            }
          }
        >
          Helvetica
        </option>
        <option
          style={
            Object {
              "fontFamily": "Garamond",
            }
          }
        >
          Garamond
        </option>
      </select>
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__color"
        data-tooltip={true}
        onChange={[Function]}
        type="color"
        value="#000000"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__button--left"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
        value="–"
      />
      <input
        className="option__number"
        data-tooltip={true}
        max={60}
        min={6}
        readOnly={true}
        type="text"
        value="20"
      />
      <input
        className="option__button option__button--right"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
        value="+"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        checked={true}
        className="option__checkbox option__checkbox--bold"
        id="bold"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="option__label"
        data-tooltip={true}
        htmlFor="bold"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        checked={false}
        className="option__checkbox option__checkbox--italic"
        id="italic"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="option__label"
        data-tooltip={true}
        htmlFor="italic"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        checked={false}
        className="option__checkbox option__checkbox--center"
        id="center"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="option__label"
        data-tooltip={true}
        htmlFor="center"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        checked={false}
        className="option__checkbox option__checkbox--uppercase"
        id="uppercase"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="option__label"
        data-tooltip={true}
        htmlFor="uppercase"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        checked={false}
        className="option__checkbox option__checkbox--underline"
        id="underline"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="option__label"
        data-tooltip={true}
        htmlFor="underline"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__down option__button--padding"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
      />
      <form
        className="option__drop-down"
        name="padding"
        onMouseLeave={[Function]}
      >
        <div
          className="option__drop-down-line"
        >
          <span>
            padding top
          </span>
          <input
            className="option__button option__button--left"
            onClick={[Function]}
            type="button"
            value="–"
          />
          <input
            className="option__number"
            max={100}
            min={0}
            readOnly={true}
            type="text"
            value={0}
          />
          <input
            className="option__button option__button--right"
            onClick={[Function]}
            type="button"
            value="+"
          />
        </div>
        <div
          className="option__drop-down-line"
        >
          <span>
            padding right
          </span>
          <input
            className="option__button option__button--left"
            onClick={[Function]}
            type="button"
            value="–"
          />
          <input
            className="option__number"
            max={100}
            min={0}
            readOnly={true}
            type="text"
            value={0}
          />
          <input
            className="option__button option__button--right"
            onClick={[Function]}
            type="button"
            value="+"
          />
        </div>
        <div
          className="option__drop-down-line"
        >
          <span>
            padding left
          </span>
          <input
            className="option__button option__button--left"
            onClick={[Function]}
            type="button"
            value="–"
          />
          <input
            className="option__number"
            max={100}
            min={0}
            readOnly={true}
            type="text"
            value={1}
          />
          <input
            className="option__button option__button--right"
            onClick={[Function]}
            type="button"
            value="+"
          />
        </div>
        <div
          className="option__drop-down-line"
        >
          <span>
            padding bottom
          </span>
          <input
            className="option__button option__button--left"
            onClick={[Function]}
            type="button"
            value="–"
          />
          <input
            className="option__number"
            max={100}
            min={0}
            readOnly={true}
            type="text"
            value={0}
          />
          <input
            className="option__button option__button--right"
            onClick={[Function]}
            type="button"
            value="+"
          />
        </div>
      </form>
    </div>
  </div>
  <div
    className="option-panel__group option-panel__group--blocks"
  >
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        checked={false}
        className="option__checkbox option__checkbox--lock"
        id="lock"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="option__label"
        data-tooltip={true}
        htmlFor="lock"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__button--autosize"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__button--back"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__button--copy"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__down option__button--link"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="option__drop-down"
      >
        <div
          className="option__drop-down-line"
        >
          <span>
            Link:
          </span>
          <input
            className="option__text option__link"
            defaultValue=""
            type="text"
          />
          <button
            className="option__button option__button--ok"
            onClick={[Function]}
          />
          <button
            className="option__button option__button--del"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__button--save"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
      />
    </div>
  </div>
</div>
`;

exports[`OptionPanel render&actions Изменение цвета текста 1`] = `
<div
  className="option-panel"
>
  <div
    className="option-panel__group option-panel__group--style"
  >
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <select
        className="option__select"
        data-tooltip={true}
        onChange={[Function]}
        value="Roboto"
      >
        <option
          style={
            Object {
              "fontFamily": "PTSans",
            }
          }
        >
          PTSans
        </option>
        <option
          style={
            Object {
              "fontFamily": "Roboto",
            }
          }
        >
          Roboto
        </option>
        <option
          style={
            Object {
              "fontFamily": "Helvetica",
            }
          }
        >
          Helvetica
        </option>
        <option
          style={
            Object {
              "fontFamily": "Garamond",
            }
          }
        >
          Garamond
        </option>
      </select>
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__color"
        data-tooltip={true}
        onChange={[Function]}
        type="color"
        value="#000000"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__button--left"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
        value="–"
      />
      <input
        className="option__number"
        data-tooltip={true}
        max={60}
        min={6}
        readOnly={true}
        type="text"
        value="20"
      />
      <input
        className="option__button option__button--right"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
        value="+"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        checked={true}
        className="option__checkbox option__checkbox--bold"
        id="bold"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="option__label"
        data-tooltip={true}
        htmlFor="bold"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        checked={false}
        className="option__checkbox option__checkbox--italic"
        id="italic"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="option__label"
        data-tooltip={true}
        htmlFor="italic"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        checked={false}
        className="option__checkbox option__checkbox--center"
        id="center"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="option__label"
        data-tooltip={true}
        htmlFor="center"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        checked={false}
        className="option__checkbox option__checkbox--uppercase"
        id="uppercase"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="option__label"
        data-tooltip={true}
        htmlFor="uppercase"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        checked={false}
        className="option__checkbox option__checkbox--underline"
        id="underline"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="option__label"
        data-tooltip={true}
        htmlFor="underline"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__down option__button--padding"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
      />
      <form
        className="option__drop-down"
        name="padding"
        onMouseLeave={[Function]}
      >
        <div
          className="option__drop-down-line"
        >
          <span>
            padding top
          </span>
          <input
            className="option__button option__button--left"
            onClick={[Function]}
            type="button"
            value="–"
          />
          <input
            className="option__number"
            max={100}
            min={0}
            readOnly={true}
            type="text"
            value={0}
          />
          <input
            className="option__button option__button--right"
            onClick={[Function]}
            type="button"
            value="+"
          />
        </div>
        <div
          className="option__drop-down-line"
        >
          <span>
            padding right
          </span>
          <input
            className="option__button option__button--left"
            onClick={[Function]}
            type="button"
            value="–"
          />
          <input
            className="option__number"
            max={100}
            min={0}
            readOnly={true}
            type="text"
            value={0}
          />
          <input
            className="option__button option__button--right"
            onClick={[Function]}
            type="button"
            value="+"
          />
        </div>
        <div
          className="option__drop-down-line"
        >
          <span>
            padding left
          </span>
          <input
            className="option__button option__button--left"
            onClick={[Function]}
            type="button"
            value="–"
          />
          <input
            className="option__number"
            max={100}
            min={0}
            readOnly={true}
            type="text"
            value={1}
          />
          <input
            className="option__button option__button--right"
            onClick={[Function]}
            type="button"
            value="+"
          />
        </div>
        <div
          className="option__drop-down-line"
        >
          <span>
            padding bottom
          </span>
          <input
            className="option__button option__button--left"
            onClick={[Function]}
            type="button"
            value="–"
          />
          <input
            className="option__number"
            max={100}
            min={0}
            readOnly={true}
            type="text"
            value={0}
          />
          <input
            className="option__button option__button--right"
            onClick={[Function]}
            type="button"
            value="+"
          />
        </div>
      </form>
    </div>
  </div>
  <div
    className="option-panel__group option-panel__group--blocks"
  >
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        checked={false}
        className="option__checkbox option__checkbox--lock"
        id="lock"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="option__label"
        data-tooltip={true}
        htmlFor="lock"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__button--autosize"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__button--back"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__button--copy"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__down option__button--link"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="option__drop-down"
      >
        <div
          className="option__drop-down-line"
        >
          <span>
            Link:
          </span>
          <input
            className="option__text option__link"
            defaultValue=""
            type="text"
          />
          <button
            className="option__button option__button--ok"
            onClick={[Function]}
          />
          <button
            className="option__button option__button--del"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
    <div
      className="option"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <input
        className="option__button option__button--save"
        data-tooltip={true}
        onClick={[Function]}
        type="button"
      />
    </div>
  </div>
</div>
`;
